src/test/java/net/minecraftforge/debug/entity/player/PlayerLoginEventTest.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2022.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.debug.entity.player;
+
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraftforge.event.entity.player.PlayerEvent;
+import net.minecraftforge.eventbus.api.Event;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+@Mod("player_login_event_test")
+@Mod.EventBusSubscriber()
+public class PlayerLoginEventTest
+{
+    private static final Logger LOGGER = LogManager.getLogger(PlayerLoginEventTest.class);
+    private static final boolean ENABLED = false;
+
+    @SubscribeEvent
+    public static void onPlayerChangeGameModeEvent(PlayerEvent.PrePlayerLoginEvent event)
+    {
+        LOGGER.info("{} logged in with connection {}.", event.getPlayer().getName(), event.getConnection());
+        if (ENABLED) {
```

Comment:
Formatting: Braces on new lines.
And to properly disable the mod, you should not even register to the event bus (i.e. you should not use `@EventBusSubscriber`).

Reply:
The ENABLED boolean is not used for disabling the test. It's used for testing if the kick works or not.

Reply:
Then it is still wrong. It _should_ disable the test.

Reply:
It's not how all other test mods work, and as I stated originally that specific boolean is for testing the kick handling (i.e. not always wanting that enabled). 

In addition to this, your initial comment was "And to properly disable the mod, you should not even register to the event bus" but in every other instance of the ENABLED boolean being used it is done exactly how I've implemented it (i.e. it is registered to the bus)

Reply:
Added global ENABLED variable and refactored old variable to KICK_ENABLED

Reply:
Why are the two variables necessary?

Reply:
Well I figured it'd probably be best to test the two cases (using the variable). So that is checking the event fires regardless of cancellation, and kick status, and then also checking the disconnect works properly. 

Can be removed if you think it's unnecessary

Reply:
It's fine I guess. The method just has the wrong name and there is still some formatting problems here.

Reply:
Updated name and formatting